### PR TITLE
handle object id's as list in Download Command

### PR DESCRIPTION
**Pull request by ayang99**

### CODE REVIEW

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/cli/ObjectIdListValidator.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+public class ObjectIdListValidator implements IValueValidator<List<String>> {
+
+  @Override
+  public void validate(String name, List<String> objectIds) throws ParameterException {
+    if (objectIds == null) {
+      return;
+    }
+
+    for (val id : objectIds) {
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/cli/ObjectIdListValidator.java]: `objectId`

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 2]
```diff
@@ -68,8 +68,8 @@
   private boolean force = false;
   @Parameter(names = "--manifest", description = "Path to manifest id, url or file")
   private ManifestResource manifestResource;
-  @Parameter(names = "--object-id", description = "Object id to download", validateValueWith = ObjectIdValidator.class)
-  private String objectId;
+  @Parameter(names = "--object-id", description = "Object id to download", validateValueWith = ObjectIdListValidator.class, variableArity = true)
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: Are you sure the previous version wouldn't work with `variableArity = true`? You shouldn't have to do anything special with lists.

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 3]
```diff
@@ -68,8 +68,8 @@
...
+  private List<String> objectId = new ArrayList<>();
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: `objectIds`

**Comment by btiernay** [comment]
@ayang99  Please address the few comments so we can merge and release. Much thanks.

**Comment by btiernay** [comment]
@ayang99  Can you please comment? Are you getting these emails?

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java (hunk 3 above)
Comment (ayang99) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: I explicitly kept the parameter the same so existing workflows wouldn't have to change. I didn't think it was worth the clutter of having separate --objectId and --objectIds parameters when it's exactly the same code

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java (hunk 2 above)
Comment (ayang99) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: The Validator is looking to parse each value to make sure it's a valid UUID. The validation method is explicitly looking for a String, not an ArrayList

**Comment by btiernay** [comment]
Please update the release notes and do a release. Thanks!

**Pull request merged**